### PR TITLE
lammps: migrate from homebrew/science

### PR DESCRIPTION
**Pull request by fxcoudert**

### CODE REVIEW

File: Formula/lammps.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+class Lammps < Formula
+  desc "Molecular Dynamics Simulator"
+  homepage "http://lammps.sandia.gov"
+  url "http://lammps.sandia.gov/tars/lammps-11Aug17.tar.gz"
+  # lammps releases are named after their release date. We transform it to
+  # YYYY.MM.DD (year.month.day) so that we get a comparable version numbering (for brew outdated).
+  # We only track "stable" releases as announced on the LAMMPS homepage.
+  version "2017.08.11"
```
Comment (ilovezfs) [inline Formula/lammps.rb]: I think we usually format these with `-` not `.`

File: Formula/lammps.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  sha256 "33431329fc735fb12d22ed33399235ef9506ba759a281a24028de538822af104"
+  revision 2
+
+  depends_on "fftw"
+  depends_on "jpeg"
+  depends_on "libpng"
+  depends_on :fortran
+  depends_on :mpi => [:cxx, :f90, :recommended]
```
Comment (ilovezfs) [inline Formula/lammps.rb]: why the option

File: Formula/lammps.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+        # Disable some packages for which we do not have dependencies, that are
+        # deprecated or require too much configuration.
+        %w[gpu kim kokkos mscg meam mpiio poems reax voronoi].each do |package|
+          system "make", "no-#{package}"
+        end
+
+        system "make", variant,
+                       "LMP_INC=-DLAMMPS_GZIP",
+                       "FFT_INC=-DFFT_FFTW3 -I#{Formula["fftw"].opt_prefix}/include",
```
Comment (ilovezfs) [inline Formula/lammps.rb]: opt_include

File: Formula/lammps.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+                       "FFT_PATH=-L#{Formula["fftw"].opt_prefix}/lib",
```
Comment (ilovezfs) [inline Formula/lammps.rb]: opt_lib

File: Formula/lammps.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+                       "FFT_LIB=-lfftw3",
+                       "JPG_INC=-DLAMMPS_JPEG -I#{Formula["jpeg"].opt_prefix}/include -DLAMMPS_PNG -I#{Formula["libpng"].opt_prefix}/include",
+                       "JPG_PATH=-L#{Formula["jpeg"].opt_prefix}/lib -L#{Formula["libpng"].opt_prefix}/lib",
+                       "JPG_LIB=-ljpeg -lpng"
+
+        bin.install "lmp_#{variant}"
```
Comment (ilovezfs) [inline Formula/lammps.rb]: have you requested a make install target

File: Formula/lammps.rb [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      end
+    end
+
+    pkgshare.install(%w[doc potentials tools bench examples])
```
Comment (ilovezfs) [inline Formula/lammps.rb]:
```
pkgshare.install "bench", "doc", "examples", "potentials", "tools"
```

Is all of this really required for the test to work?

File: Formula/lammps.rb [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  end
+
+  test do
+    system "#{bin}/lmp_serial", "-in", "#{HOMEBREW_PREFIX}/share/lammps/bench/in.lj"
```
Comment (ilovezfs) [inline Formula/lammps.rb]: pkgshare

File: Formula/lammps.rb [hunk 8]
```diff
@@ -0,0 +1,49 @@
+class Lammps < Formula
+  desc "Molecular Dynamics Simulator"
+  homepage "http://lammps.sandia.gov"
+  url "http://lammps.sandia.gov/tars/lammps-11Aug17.tar.gz"
+  # lammps releases are named after their release date. We transform it to
```
Comment (ilovezfs) [inline Formula/lammps.rb]: It would be nice to make a request upstream that they name their tarballs sanely

File: Formula/lammps.rb [hunk 9]
```diff
@@ -0,0 +1,49 @@
+class Lammps < Formula
+  desc "Molecular Dynamics Simulator"
+  homepage "http://lammps.sandia.gov"
```
Comment (ilovezfs) [inline Formula/lammps.rb]: trailing slash

File: Formula/lammps.rb [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+        %w[gpu kim kokkos mscg meam mpiio poems reax voronoi].each do |package|
+          system "make", "no-#{package}"
+        end
+
+        system "make", variant,
+                       "LMP_INC=-DLAMMPS_GZIP",
+                       "FFT_INC=-DFFT_FFTW3 -I#{Formula["fftw"].opt_prefix}/include",
+                       "FFT_PATH=-L#{Formula["fftw"].opt_prefix}/lib",
+                       "FFT_LIB=-lfftw3",
+                       "JPG_INC=-DLAMMPS_JPEG -I#{Formula["jpeg"].opt_prefix}/include -DLAMMPS_PNG -I#{Formula["libpng"].opt_prefix}/include",
```
Comment (ilovezfs) [inline Formula/lammps.rb]: opt_include

File: Formula/lammps.rb [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+                       "JPG_PATH=-L#{Formula["jpeg"].opt_prefix}/lib -L#{Formula["libpng"].opt_prefix}/lib",
```
Comment (ilovezfs) [inline Formula/lammps.rb]: opt_lib

File: Formula/lammps.rb [hunk 12]
```diff
@@ -0,0 +1,49 @@
+class Lammps < Formula
+  desc "Molecular Dynamics Simulator"
+  homepage "http://lammps.sandia.gov"
+  url "http://lammps.sandia.gov/tars/lammps-11Aug17.tar.gz"
+  # lammps releases are named after their release date. We transform it to
+  # YYYY.MM.DD (year.month.day) so that we get a comparable version numbering (for brew outdated).
+  # We only track "stable" releases as announced on the LAMMPS homepage.
```
Comment (ilovezfs) [inline Formula/lammps.rb]: The quotes aren't needed

File: Formula/lammps.rb (hunk 5 above)
Comment (fxcoudert) [reply Formula/lammps.rb]: They're moving to cmake (❤️). I'll make a request anyway, but in a few months this should all be gone!

File: Formula/lammps.rb (hunk 6 above)
Comment (fxcoudert) [reply Formula/lammps.rb]: potentials and tools are required.
doc is nice to have (and not that big).
bench is where I get the example files from
examples we could do without, but I think they're linked to the docs.

Comment (ilovezfs) [reply Formula/lammps.rb]: ok

**Pull request closed without merging**